ui/common/ContributionBar.tsx

```diff
@@ -0,0 +1,72 @@
+'use client'
+
+import Image from 'next/image'
+
+interface ContributionInfo {
+  username: string
+  avatar: string
+  color: string
+  percentage: number
+  side: number
+  zindex: string
+}
+
+interface ContributionInfoProps {
+  contributionInfo: ContributionInfo[]
+}
+
+export default function ContributionBar({
+  contributionInfo,
+}: ContributionInfoProps) {
+  let value = 0
+  return (
+    <div className="relative h-[30px] w-full rounded-[5px] bg-black/20">
+      <div className="absolute z-50 flex h-full w-1/2 border-r-2 border-dotted border-white/50"></div>
+      {contributionInfo.map((info, i) => {
+        value += info.percentage
+        return (
+          <>
+            <div
+              className={`absolute h-full w-full overflow-hidden rounded-[5px] `}
+            >
+              <div
+                className={`absolute inset-0 flex transform-gpu border-r-2 border-black transition-transform`}
+                style={{
+                  transform: `translate3d(${
+                    info.side === 0 ? 100 - info.percentage : -(100 - value)
+                  }%,0,0)`,
+                  backgroundColor: `#${info.color}`,
+                  zIndex: info.zindex,
```

Comment:
Not critical but could this just be handled by checking the index inherent to the map like `zIndex: {i * 10}`

Reply:
Yes, we can do that. The unusual balance of avatar position is due to the value provided by the parent component. It can adjusted.